src/objects.rs

```diff
@@ -207,7 +429,43 @@ pub struct DeciderConditions {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub output_signal: Option<SignalID>,
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub copy_count_from_input: Option<bool>
+    pub copy_count_from_input: Option<bool>,
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, Deserialize, Serialize)]
+pub struct LogisticCondition {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub first_signal: Option<SignalID>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub second_signal: Option<SignalID>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub constant: Option<i32>,
+    pub comparator: String,
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, Deserialize_repr, Serialize_repr)]
+#[repr(u32)]
+pub enum CircuitModeOfOperation {
+    /// TODO what does this correspond to? power_book.txt has this set to 0
+    RoboportThing = 0,
+    SetRequests = 1,
+    /// TODO what does this correspond to?
+    // TWO = 2,
+    None = 3,
+}
```

Comment:
I understand that this is the result of reverse-engineering the format, but when there are free-floating TODOs like this, I'd like to see them link to an issue in the repo.